src/room.ts

```diff
@@ -14,9 +14,9 @@ type RoomObj = {
   encryId:    string,
   topic:      string,
   ownerUin:   number,
-  memberList: Contact[],
-  nameMap:    Map<string, string>,
-  aliasMap:   Map<string, string>,
+  memberList?: Contact[],
+  nameMap?:    Map<string, string>,
```

Comment:
Why I said "right" is because you starting to notice the right problem.

But what you did modify here is not correct.

the `memberList`, `nameMap` and `aliasMap` should not defined by `?`, which means they should always not to be `undefined` or `null`.

If there's no data for them, they should be empty array, or a empty `{}`.